namespaces/resource-namespace.md

```diff
@@ -0,0 +1,150 @@
+
+[MSON][] is used throughout this document.
+
+## Supporting Data Types
+
+### Href (string)
+
+The value of `Href` is to be used as a URI Template, as defined in RFC 6570.
+The value of the `Href` is a template used to determine the target URI of the related resource.
+The value of the instance property SHOULD be resolved as a URI-Reference per RFC 3986 and MAY be a relative reference to a URI.
+
+### Href Variables (Object Type)
+
+The definition is a MSON namespace `Object Type` where keys are respective URI Template variables.
+
+### Data Structure (MSON Element)
+
+Data structure definition using MSON namespace elements.
+
+## Namespace Elements
+
+### Resource (Element)
+
+The Resource representation with its available transitions and attributes.
+
+#### Properties
+
+- `element`: resource (string, fixed)
+- `attributes` (object)
+    - `href` (Href) - URI Template of this resource.
+    - `hrefVariables` (Href Variables) - Definition of URI Template variables used in the `href` property.
+
+- `content` (array)
+    - (Transition) - State transition available for this resource
+    - (Data Structure) - Data structure associated with the resource
+
+#### Example
+
+```json
+{
+  "element": "resource",
+  "meta": {
+    "title": "Question",
+    "description": "A Question object has the following attributes."
+  },
+  "attributes": {
+    "href": "/questions/{question_id}",
+    "hrefVariables": {
+      "element": "object",
+      "content": [
+        {
+          "element": "member",
+          "content": {
+            "key": {
+              "element": "string",
+              "content": "question_id"
+            },
+            "value": {
+              "element": "number",
+              "content": "1"
+            }
+          }
+        }
+      ]
+    }
+  },
+  "content": [
+    {
+      "element": "object",
+      "meta": {
+        "title": "Question"
+      },
+      "content": [
+        {
+          "element": "member",
+          "content": {
+            "key": {
+              "element": "string",
+              "content": "question"
+            },
+            "value": {
+              "element": "string",
+              "content": "Favourite programming language?"
+            }
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+### Transition (Element)
+
+A transition is an available progression from one state to another state. Exercising a transition initiates HTTP transaction.
+
+#### Properties
+
+- `element`: transition (string, fixed)
+- `attributes` (object, required)
+    - `relation` - (string) - The relation of this transition from the current resource.
+- `content` (object, required)
+    - `href` (Href, required) - The URI template for this transition.
+    - `parameters` (Href Variables) - A hash of input parameters for this transition.
+    - `attributes` (Data Structure) - Input attributes for this transition.
+    - `transactions` (array[Transaction]) - Array of transaction examples.
```

Comment:
It may be helpful to address issue #12 first before committing to this, as you may be able to do what you originally had planned, which was to put `href`, `parameters`, and `attributes` in the `attributes` section, along with `content` being defined as an array of Transaction elements. Would that simplify this any?